src/core/api/__tests__/public_api.test.ts

```diff
@@ -0,0 +1,571 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { expect } from "chai";
+import PublicAPI from "../public_api";
+
+describe("API - Public API", () => {
+  describe("static properties", () => {
+    describe("ErrorTypes", () => {
+      it("should expose static ErrorTypes property", () => {
+        expect(typeof PublicAPI.ErrorTypes).to.equal("object");
+      });
+    });
+
+    describe("ErrorCodes", () => {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Update or refactor this function so that its implementation doesn't duplicate the one on line 22. ([typescript:S4144](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4144))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=437&issues=AWgOIuUCUMkE2J58176D&open=AWgOIuUCUMkE2J58176D)

[ISSUE_KEY]: AWgOIuUCUMkE2J58176D